examples/interaction/tools/hover_tooltip_advanced.py

```diff
@@ -0,0 +1,56 @@
+import numpy as np
+
+from bokeh.models import HoverTool, Styles
+from bokeh.models.dom import Div, Index, ValueRef
+from bokeh.palettes import Spectral11
+from bokeh.plotting import figure, show
+
+N = 1000
+x = np.random.random(size=N) * 100
+y = np.random.random(size=N) * 100
+radii = np.random.random(size=N) * 1.5
+colors = np.random.choice(Spectral11, size=N)
+
+p = figure(
+    title="Demonstrates hover tool with advanced and regular tooltip formatting side-by-side",
+    tools="pan,wheel_zoom,box_select,crosshair",
+)
+
+p.circle(x, y, radius=radii, fill_color=colors, fill_alpha=0.6, line_color=None)
+
+grid = Div(
+    style=Styles(
+        display="grid",
+        grid_template_columns="auto auto",
+        column_gap="10px",
+    ),
+    children=[
+        "index:",  Div(children=["#", Index()]),
+        "(x, y):", Div(children=["(", ValueRef(field="x"), ", ", ValueRef(field="y"), ")"]),
```

Comment:
Just thinking out loud here, having a convenience class like this could make sense to me. 

```python
        "(x, y):", Div(children=[ValueRefs("({x}, {y})", fields=["x", "y"])]),
```
or
``` python
        "(x, y):", ValueRefs("({x}, {y})", fields=["x", "y"]),
```
Likely out-of-scope for this PR.

Reply:
`ValueRef` is a low-level API meant for having full control over formatting, instead of relying on an ad-hoc syntax. However, most of the time users will choose the syntax out of convenience. Thus I don't see a need for additional convenience here. If only PEP 501 (general purpose string interpolation) wasn't deferred, then this could be made really convenient while retaining complete control.